basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/ElytronSslContextTestCase.java

```diff
@@ -0,0 +1,595 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.jboss.arquillian.graphene.page.Page;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.dmr.ModelNode;
+import org.jboss.hal.testsuite.category.Shared;
+import org.jboss.hal.testsuite.creaper.ResourceVerifier;
+import org.jboss.hal.testsuite.dmr.ModelNodeGenerator.ModelNodePropertiesBuilder;
+import org.jboss.hal.testsuite.dmr.ModelNodeGenerator.ModelNodeListBuilder;
+import org.jboss.hal.testsuite.fragment.formeditor.Editor;
+import org.jboss.hal.testsuite.fragment.shared.modal.WizardWindow;
+import org.jboss.hal.testsuite.fragment.shared.modal.WizardWindowWithOptionalFields;
+import org.jboss.hal.testsuite.page.config.elytron.SSLPage;
+import org.jboss.hal.testsuite.util.ConfigChecker;
+import org.jboss.hal.testsuite.util.ElytronIntegrationChecker;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class ElytronSslContextTestCase extends AbstractElytronTestCase {
+
+    private static final String
+        KEY_STORE = "key-store",
+        CLEAR_TEXT = "clear-text",
+        TYPE = "type",
+        JKS = "jks",
+        CREDENTIAL_REFERENCE = "credential-reference",
+        ALGORITH_VALUE_1 = "PKIX",
+        ALGORITH_VALUE_2 = "SunX509",
+        KEY_MANAGER_LABEL = "Key Manager",
+        KEY_MANAGERS = "key-managers",
+        TRUST_MANAGER_LABEL = "Trust Manager",
+        TRUST_MANAGERS = "trust-managers",
+        ALGORITHM = "algorithm",
+        CREDENTIAL_REFERENCE_CLEAR_TEXT_IDENTIFIER = "credential-reference-clear-text",
+        ALIAS_FILTER = "alias-filter",
+        PROVIDERS = "providers",
+        CREDENTIAL_REFERENCE_LABEL = "Credential Reference",
+        SERVER_SSL_CONTEXT = "server-ssl-context",
+        SERVER_SSL_CONTEXT_LABEL = "Server SSL Context",
+        CLIENT_SSL_CONTEXT = "client-ssl-context",
+        CLIENT_SSL_CONTEXT_LABEL = "Client SSL Context",
+        TLS_V11 = "TLSv1.1",
+        TLS_V12 = "TLSv1.2",
+        PROTOCOLS = "protocols",
+        NEED_CLIENT_AUTH = "need-client-auth",
+        SESSION_TIMEOUT = "session-timeout";
+
+    @Page
+    private SSLPage page;
+
+    /**
+     * @tpTestDetails Try to create Elytron Key Manager instance in Web Console's Elytron subsystem configuration.
+     * Validate created resource is visible in Key Manager table.
+     * Validate created resource is in model.
+     * Validate attribute values of created resource in model.
+     */
+    @Test
+    public void addKeyManagerTest() throws Exception {
+        Address keyStoreAddress = createKeyStore();
+        String keyManagerName = randomAlphanumeric(5), password = randomAlphanumeric(5),
```

Comment:
Will be resources with name consisting just of random string identifiable in case of debugging?